snuba/admin/views.py

```diff
@@ -81,20 +83,35 @@ def health() -> Response:
 
 @application.route("/migrations/groups")
 def migrations_groups() -> Response:
-    res: List[Mapping[str, MigrationGroup | Sequence[str]]] = []
-    for migration_group in get_active_migration_groups():
-        if migration_group.value in settings.ADMIN_ALLOWED_MIGRATION_GROUPS:
-            group_migrations = get_group_loader(migration_group).get_migrations()
-            res.append(
-                {"group": migration_group.value, "migration_ids": group_migrations}
-            )
+    res: List[Mapping[str, str | Sequence[Mapping[str, str | bool]]]] = []
+    allowed_groups = [
+        group.value
+        for group in get_active_migration_groups()
+        if group.value in settings.ADMIN_ALLOWED_MIGRATION_GROUPS
+    ]
+    if not allowed_groups:
+        return make_response(jsonify(res), 200)
+
+    for group, migrations in runner.show_all(allowed_groups):
```

Comment:
Seems like a lot of the correctness of this algo depends on the show_all returning things in order. Do we know if that is guaranteed? Maybe we can add a line in docstring showing if that's the case.

Reply:
so looked into this, and when doing the status checks in the runner we look at the `get_migrations()` from the loader, which makes sense since that's where the order is explicitly defined. I've updated the logic to use to loader so we can make sure the ordering is consistent